### PR TITLE
Cleanup productdb info lookup

### PR DESCRIPTION
**Pull request by karlpauls**
Test URLs:
- Before: https://main--mammotome--hlxsites.hlx.page/us/en/
- After: https://cleanup-productdb-info--mammotome--hlxsites.hlx.page/us/en/

### CODE REVIEW

**Comment by solaris007** [comment]
we do have a consistent TBT - is this related to the product db or something else?

File: scripts/lib-franklin.js [hunk 1]
```diff
@@ -366,13 +366,43 @@ export function decorateSupScript(string, result = []) {
   return result;
 }
 
+export function getInfo() {
+  const [, country, language] = window.location.pathname.split('/');
+  return {
+    country,
+    language,
+    productDB: new URL('/products.json', window.location).pathname,
```
Comment (solaris007) [inline scripts/lib-franklin.js:374]: i would move the URL constructions into a small local-scope helper function for readability and easier adjustment:

```
  const getUrlPath = (path) => new URL(path, origin).pathname;
  
  return {
    country,
    language,
    productDB: getUrlPath('/products.json'),
    productSupport: getUrlPath(`/${country}/${language}/product-support`),
    queryIndex: getUrlPath(`/${country}/${language}/query-index.json`),
  };
```

File: scripts/lib-franklin.js [hunk 2]
```diff
@@ -366,13 +366,43 @@ export function decorateSupScript(string, result = []) {
...
+    productSupport: new URL(`/${country}/${language}/product-support`, window.location).pathname,
+    queryIndex: new URL(`/${country}/${language}/query-index.json`, window.location).pathname,
+  };
+}
+
+export function adjustAssetURL(asset) {
+  if (asset?.URL) {
+    const url = new URL(asset.URL, window.location);
+    if (url.hostname.endsWith('-mammotome--hlxsites.hlx.page')
```
Comment (solaris007) [inline scripts/lib-franklin.js:383]: you could shorten this slightly by having an array of host names. the array values could even be a constant somewhere. using an array would make it easier to add/remove domains:
```
    const isLocalOrHlx = ['localhost', '-mammotome--hlxsites.hlx.page', '-mammotome--hlxsites.hlx.live']
      .some(domain => url.hostname.endsWith(domain));
      
    if (isLocalOrHlx) {
      asset.URL = url.pathname;
    }
```

File: scripts/lib-franklin.js [hunk 3]
```diff
@@ -366,13 +366,43 @@ export function decorateSupScript(string, result = []) {
...
 export async function getProductDB() {
   if (!window.productDB) {
-    const resp = await fetch('/products.json?limit=10000');
+    const { productDB } = getInfo();
+    const resp = await fetch(`${productDB}?limit=10000`);
     if (!resp.ok) {
       throw new Error(`${resp.status}: ${resp.statusText}`);
     }
     window.productDB = await resp.json();
+    if (window.productDB.ProductAsset?.data) {
```
Comment (solaris007) [inline scripts/lib-franklin.js:400]: in the same vein, if we like arrays and functional, we could make this slightly more maintainable:

```
    const adjustData = (field) => {
      if (window.productDB[field]?.data) {
        window.productDB[field].data = window.productDB[field].data.map(adjustAssetURL);
      }
    };

    ['ProductAsset', 'eIFU'].forEach(adjustData);
  }
  
  return window.productDB;
```

**Comment by karlpauls** [comment]
@solaris007 , I can't imagine how it could be. It looks like this is an artifact of the martec maybe?

**Review by solaris007 — APPROVED** [review]

**Pull request merged**